### DISCUSSION TITLE
Can UI react to incoming MIDI

### DISCUSSION THREAD

**Discussion by adonaldson** [2xAA/genmdm-editor#138] [Ideas]
I created a very basic MIDI merger using MIDImonster which merged incoming note data with MIDI CCs from my Akai Midimix.

Configuring the editor to take in this new stream and output to my megadrive allowed me to play the megadrive and tweak settings either in the editor or by twiddling knobs on the Midimix.

Is it possible to have the UI react to incoming CCs, updating to give feedback on the parameter changes?

Thanks for all your hard work. I'm happy to help out, just curious to know whether it's remotely possible without bigger structural changes!

**Comment by 2xAA** [comment]
Hi @adonaldson!

We had a PR to add MIDI CC interface updates a while ago (v1.4.0).

Are you using the latest version?

I also tested this just the other week using a MIDI controller and it still seemed to work

**Reply by adonaldson** [reply]
Hey 2xAA, thanks for getting back. I'll give this a whirl later and let you know how you get on - I'm pretty sure I'm up to date.

With my setup I'm able to affect genmdm with my changes through the editor using my MIDI controller as though I'm touching the UI in the editor itself.

However, the editor UI itself doesn't react to the incoming CC, so doesn't stay in sync with what data is sent to the megadrive. But let me double check this and get back to you, it's very possible my janky setup is introducing problems :)

Cheers!

Andrew

**Reply by 2xAA** [reply]
Hey Andrew, no worries!

Please let me know how you get on.

This is the commit with the changes to handle incoming CC to update the UI:
https://github.com/2xAA/genmdm-editor/commit/96b3b506b3204068c5d31c03a31a604fa93f6942#diff-7a7a37b12ee1265d7e27577ec286120d2cbc0940908635e264a2be44ccb9a8a0R517

You can see this at App.vue:517 and App.vue:607.

With Vue.js, updating the CC values should reflect directly in the UI.
I'd monitor your MIDI CC on the input of GenMDM Editor. Check which channel is being updated.

MIDI channel 1-6 will only show the UI updating if the "channel" parameter in the editor matches the incoming MIDI.
If you're in polyphony mode in the editor, any MIDI channel will update the UI.

**Reply by adonaldson** [reply]
You're absolutely right of course - the problem was entirely at my end.

So for anyone coming later, the answer to the question is **yes the editor UI will update itself based on incoming CCs**.

Thanks again for your time!
